### PR TITLE
Fixed Medication Update Issue

### PR DESCRIPTION
**Pull request by MonishaSivanathan**
- changed the optional field
- changed the logic for update and hadled all scenario

### CODE REVIEW

File: controller/medication_controller.py [hunk 1]
```diff
@@ -140,78 +140,114 @@ def update_medication_by_patient_id(patient_id: str, med: MedicationUpdateModel)
...
-                    intent=INTEND,
-                    note=[Annotation(text=OTHER)],
+                    status=current_status,
+                    note=[Annotation(text=CURRENT)],
                 )
-                medication_request.save()
-                result["medication_request_created_id"] = medication_request.id
+                create_med_statement.save()
+                result["medication_statement_update_id"] = create_med_statement.id
             else:
```
Comment (tamsud) [inline controller/medication_controller.py:178]: here i think we need to make it as a elif, need clarfication. can you expalin all the scenarios

File: controller/medication_controller.py [hunk 2]
```diff
@@ -140,78 +140,114 @@ def update_medication_by_patient_id(patient_id: str, med: MedicationUpdateModel)
...
-                    note=[Annotation(text=CURRENT)],
+                    status=history_status,
+                    intent=INTEND,
+                    note=[Annotation(text=OTHER)],
                 )
-                create_med_statement.save()
-                result["medication_statement_created_id"] = create_med_statement.id
+                medication_request.save()
+                result["medication_request_update_id"] = medication_request.id
             else:
```
Comment (tamsud) [inline controller/medication_controller.py:232]: same here

**Review by tamsud — CHANGES_REQUESTED** [review]

File: controller/medication_controller.py (hunk 1 above)
Comment (MonishaSivanathan) [reply controller/medication_controller.py:178]: i have changed

File: controller/medication_controller.py (hunk 2 above)
Comment (MonishaSivanathan) [reply controller/medication_controller.py:241]: changed

**Review by tamsud — APPROVED** [review]

**Pull request merged**